src/ucp/api/ucp.h

```diff
@@ -2006,6 +2031,30 @@ ucs_status_ptr_t ucp_ep_flush_nb(ucp_ep_h ep, unsigned flags,
                                  ucp_send_callback_t cb);
 
 
+/**
+ * @ingroup UCP_ENDPOINT
+ *
+ * @brief Non-blocking flush of outstanding AMO and RMA operations on the
+ * @ref ucp_ep_h "endpoint".
+ *
+ * This routine flushes all outstanding AMO and RMA communications on the
+ * @ref ucp_ep_h "endpoint". All the AMO and RMA operations issued on the
+ * @a ep prior to this call are completed both at the origin and at the target
+ * @ref ucp_ep_h "endpoint" when this call returns.
+ *
+ * @param [in] ep        UCP endpoint.
+ * @param [in] param     Operation parameters, see @ref ucp_request_param_t.
```

Comment:
pls document which flags could be specified in param->flags

Reply:
in current implementation flags are not used, will update docs after new functionality added

Reply:
as discussed pls add flags

Reply:
pls add note about operation-specific flags

Reply:
added note

Reply:
@yosefe I understand you request differently. Seems like you explicitly suggest to least the flags that are relevant for this operations. I don't think this has been done.

Reply:
@shamisp this is now added in lines 2049-2051